core/server/worker/src/main/java/alluxio/worker/block/allocator/MaxFreeAllocator.java

```diff
@@ -119,6 +96,33 @@ private StorageDirView allocateBlock(long sessionId, long blockSize,
     return candidateDirView;
   }
 
+  /**
+   * Find available dir in tiers for a block with blockSize.
+   *
+   * @param blockSize  the requested block size
+   * @param skipReview wether need to review
+   * @param mediumType the medium type to find a dir
+   * @return available dir; null if fail to find a dir
+   */
+  public StorageDirView getAvailDirInAnyTier(long blockSize, boolean skipReview,
+                                             String mediumType) {
+    StorageDirView finalCandidate = null;
```

Comment:
In the spirit of MaxFree allocation, I believe first candidate is a better choice for when all candidates are rejected.

Reply:
> In the spirit of MaxFree allocation, I believe first candidate is a better choice for when all candidates are rejected.
@ggezer hi~
behavior now:
available block-1 in tier1 , reviewer rejected -> no available block in tier2 -> allocate null block
available block-1 in tier1 , reviewer rejected -> available block-2 in tier2 , reviewer rejected -> allocate block-2

so you think it is better that?
available block-1 in tier1 , reviewer rejected -> no available block in tier2 -> allocate  block-1 
available block-1 in tier1 , reviewer rejected -> available block-2 in tier2 , reviewer rejected -> allocate  block-1

Reply:
Exactly. Effectively, ignoring the reviewer when it rejects everything and make the allocation as if reviewer was not consulted.

Reply:
@jiacheliu3 What do you think is more appropriate behavior?  I can modify the existing implementation if you agree with that.